src/schema-association-service.ts

```diff
@@ -103,6 +108,20 @@ async function autoDetectSchema(
     if (!(await azureAccountApi.waitForSubscriptions())) {
         logger.log(`Waiting for login`, 'SchemaDetection');
 
+        // Disable 1ESPT schema and delete 1ESPT schema file if user is signed out
+        const schemaUri1ESPTFolder = Utils.joinPath(context.globalStorageUri, '1ESPTSchema');
+        var schemaUri1ESPTFolderExists = fs.existsSync(schemaUri1ESPTFolder.path.substring(1)); // substring(1) to remove the '/' at the beginning of path
+        config.update('1ESPipelineTemplatesSchemaFile', undefined, vscode.ConfigurationTarget.Global); // disable 1ESPT schema
+        logger.log("1ESPT schema disabled as user is not signed in", 'SchemaDetection')
+
+        // show message to user that 1ESPT schema is disabled as user is not signed in
+        vscode.window.showInformationMessage(Messages.disabled1ESPTSchemaAsUserNotSignedInMessage)
+
+        if(schemaUri1ESPTFolderExists){
```

Comment:
Curious to hear the rationale about deleting the cached schemas

Reply:
We don't want 1ESPT schema files to be present on machines where the users are not signed in. This is to take care of scenarios when shared machines are used